src/main/java/com/afj/solution/buyitapp/model/User.java

```diff
@@ -140,4 +141,28 @@ public boolean isEnabled() {
     public String toString() {
         return GSON.toJson(this);
     }
+
+    public User update(final User newUser, final User currentUser) {
+        final String username = isNull(newUser.username) ? this.username : newUser.username;
+        final String email = isNull(newUser.email) ? this.email : newUser.email;
+        final String firstName = isNull(newUser.firstName) ? this.firstName : newUser.firstName;
+        final String lastName = isNull(newUser.lastName) ? this.lastName : newUser.lastName;
+        final String phoneNumber = isNull(newUser.phoneNumber) ? this.phoneNumber : newUser.phoneNumber;
+        final String homeAddress = isNull(newUser.homeAddress) ? this.homeAddress : newUser.homeAddress;
+        final ZonedDateTime dateOfBirth = isNull(newUser.dateOfBirth) ? this.dateOfBirth : newUser.dateOfBirth;
+        final Set<GrantedAuthority> authorities = isNull(newUser.authorities) ? this.authorities : newUser.authorities;
+        currentUser.setFirstName(firstName);
+        currentUser.setLastName(lastName);
+        currentUser.setEmail(email);
+        currentUser.setUsername(username);
+        currentUser.setPhoneNumber(phoneNumber);
+        currentUser.setHomeAddress(homeAddress);
+        currentUser.setDateOfBirth(dateOfBirth);
+        currentUser.setAccountNonExpired(accountNonExpired);
+        currentUser.setAccountNonLocked(accountNonLocked);
```

Comment:
not need to update that field by `public User update(final User newUser, final User currentUser)` method.